nupic/research/frameworks/vernon/experiments/self_supervised_experiment.py

```diff
@@ -0,0 +1,452 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import sys
+
+import torch
+import torch.nn.functional as F
+from torch.backends import cudnn
+from torch.optim.lr_scheduler import OneCycleLR
+from torch.utils.data import DataLoader, RandomSampler
+
+from nupic.research.frameworks.pytorch.lr_scheduler import ComposedLRScheduler
+from nupic.research.frameworks.pytorch.self_supervised_utils import EncoderClassifier
+from nupic.research.frameworks.vernon.experiment_utils import create_lr_scheduler
+from nupic.research.frameworks.vernon.experiments import SupervisedExperiment
+from nupic.research.frameworks.vernon.network_utils import create_model
+
+__all__ = ["SelfSupervisedExperiment"]
+
+
+# Improves performance when using fixed size images (224) and CNN
+cudnn.benchmark = True
```

Comment:
I don't think this line is needed as it will be called when importing `SupervisedExperiment`.

Reply:
Removed